include/algorithmFunctions.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+
+function calculatePoints($event){
+
+	$points = 0;
+
+	// these both get in the format of 'July 24, 2018'
+	$shortEventDate=formatDateToCalculate($event['dateOfEvent']);
+	$shortTodayDate= date("F j, Y");
+
+	//00:00 format
+	$fullEventDate = $event['dateOfEvent'];
+	$fullTodayDate = date('Y-m-d H:i:s');
+
+	$dateDiff= dateDifference($fullTodayDate, $event['dateOfEvent'])+1;
+
+
+	//NOTE: there is probably a better way to do this than a bunch of if statements,
```

Comment:
Probably an easier way to do this would be to make an array that maps # of days to a point value. Then you could just loop through that and apply the number of points as soon as you find the appropriate row. That'd save you from having all these `else if` statements.

Reply:
Makes sense. I get how to put a single if equals statement into the array and map it to a point value (like if($dateDiff == 1), give 400 points would be 1->400), but is there anyway to put the ranges into the array too? Like if(15>$dateDiff&&$dateDiff > 5) then map to 100?

Reply:
You could do this in one of two ways. 

The way I wouldn't recommend is to put the start and end range into an array like

```
0=> array(
    'Start'=>5,
    'End'=>15,
    'Points'=>100
)
```

Probably better than that though would be to only have one value, and you assume anything lower than it gets the number of points shown. So you loop through the array until you find an entry with a higher value than the one you're looking for, and give it that number of points.